### ISSUE TITLE
KIF cannot interact with UIAlertViews

### ISSUE THREAD

**Issue by aeioliu** [kif-framework/KIF#1008]
`waitForView(withAccessibilityLabel:)` or `tapView(withAccessibilityLabel:)` can no longer interact with `UIAlertView` with an action sheet style, despite the fact that the accessibility inspector confirms that the label and accessibility label is correct.

The `ModalViewTests` also fail in the KIF test suite on Xcode 9 / iOS 11 simulator.

**Comment by justinseanmartin** [comment]
Fun. Thanks for calling this out!

I'll need to dig in when I have time, but if you have time to start investigating why this is the case, I'd definitely appreciate it.

**Comment by ristkof** [comment]
Seems on 11.2 now this is for every UIButton the case (also UINavigationItem for instance). In 11.1 things work ok.

**Comment by mikelupo** [comment]
Works ok in my 11.2 devices if my own app displayed the alert.

**Comment by ristkof** [comment]
Just to be clear:
- alerts and buttons do not work in 11.2 simulator
- alerts and buttons do work in 11.1 simulator

I haven't tried on device yet. I need to look into more detail, but KIF fails to find a UIButton based on accessibilityLabel in 11.2 simulator.

**Comment by justinseanmartin** [comment]
You're saying you're unable to tap any button with the iOS 11.2 simulator?

**Comment by justinseanmartin** [comment]
I just opened `KIF.xcodeproj` with Xcode 9.2 on the iPhone 8 Plus w/ iOS 11.2 and ran the integration test suite for KIF and they're passing (with the exception of the `BackgroundTests` and `ModalViewTests`) are passing just fine. 

Can you be more specific about what isn't working for you?

**Comment by justinseanmartin** [comment]
Leaving open for the original issue of Alerts being non-interactable, which still requires some investigation.

**Comment by ristkof** [comment]
@justinseanmartin yes that's what I'm saying, in the iPhone 7 simulator, Xcode 9.2. It's consistent that in iOS 11.2 tests fail on buttons where in iOS 11.1 they work.

I'll try and isolate it if I have time. Keep this bug closed, if I have something isolated that I can send over I'll file a new bug.

**Comment by justinseanmartin** [comment]
Are these system alerts or or alerts presented by the app?

I'm in the process of getting CI set up to run on Xcode 9.2 w/ iOS 11.2 in #1048. There are tests such as `[ModalViewTests testInteractionWithAnAlertView]` that show that general interaction with an alert view is working just fine. 

If you can give more detail or better yet add a test case showing the failing scenario, I'd be happy to give further guidance.

**Comment by ristkof** [comment]
I isolated my bug and filed a new issue here https://github.com/kif-framework/KIF/issues/1050

**Comment by justinseanmartin** [comment]
I'm going to close this issue out for now. #1050 had a different root cause I believe than what was being reported here. To get CI passing in iOS 11, I had to disable `ModalViewTests testInteractionWithAnActivityViewController`, as the activity view is now hosted in a remote window that KIF can't interact with.

@aeioliu, let me know if you have more detail on what is broken and I can try to take a look. Alternatively, adding a new integration test to the KIF project is a simple way to get issues for investigated.
